deepchecks/core/suite.py

```diff
@@ -115,6 +115,47 @@ def select_results(self, idx: Set[int]) -> List[Union[
                 output.append(result)
         return output
 
+    def select_results_by_check_kind(
```

Comment:
kind is not vary clear, Would rename to type.
Consider also allowing passing the check kind as str

Reply:
> kind is not vary clear, Would rename to type.

renamed

> Consider also allowing passing the check kind as str

it is always possible to make a mistake in a string, so I do not see how it is better/